File: leetcode/104-Maximum-Depth-of-Binary-Tree/phyllis.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    
    int findDepth(TreeNode* root, int depth){
        if(root->left == NULL && root->right == NULL)
            return depth;
        
        int rightDepth = (root->right == NULL) ? INT_MIN : findDepth(root->right,depth+1);
        int leftDepth = (root->left == NULL) ? INT_MIN : findDepth(root->left,depth+1);
        return (rightDepth < leftDepth) ? leftDepth : rightDepth;
    }
    
    int maxDepth(TreeNode* root) {
        if(!root) return 0;
        else return findDepth(root, 1);
    }
};

File: leetcode/257-Binary-Tree-Paths/phyllis.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {   
    ArrayList<String> list;
    
    public List<String> binaryTreePaths(TreeNode root) {
        list = new ArrayList<String>();
        traversal(root, "");
    }

    public String traversal(TreeNode root, String str){
        str + "->" + root.data;        
        if(root.left == NULL && root.right == NULL)
            return str;
        if(root.left != NULL) 
            return traversal(root.left, str);
        if(root.right != NULL) 
            return traversal(root.right, str);
        list.add(str);        
    }
}

